portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,70 +27,107 @@
+                String comment = (String) (entity.getProperty("newComment"));
+
+                //For testing queried result comments
+                // System.out.println("Query result comment: " + comment);
+
+                pastComments.add(comment);
+            } else {
+                System.out.println("Input of number comments wanted not greater than or equal to 0: " + validNumRequest);
+            }
+           
+            //Grab number of comments wanted to be seen by user only in the first entity
+            while (numCommentsWantedTrigger) {
+                numCommentsWanted = validNumRequest;
+                numCommentsWantedTrigger = false;
+            }
+        }
+ 
+        if (numCommentsWanted > pastComments.size()) {
+            numCommentsWanted = pastComments.size();           
         }
+ 
+        pastComments = pastComments.subList(0, numCommentsWanted);
 
-        List<String> factsAndComments =  new ArrayList<>();
-        factsAndComments.addAll(funFacts);
-        factsAndComments.addAll(pastComments);
+        // For testing proper number and right comments are in sublist
+        // for (String comment : pastComments) {
+        //     System.out.println("sublist comment: " + comment);
+        // }
 
+        factsAndComments.addAll(pastComments);
+ 
         response.setContentType("application/json;");
         Gson gson = new Gson();
         response.getWriter().println(gson.toJson(factsAndComments));
     }
-
+ 
     /** Process POST request form input, add to pastComments, and redirect to index.html */
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        // Get the input from the form.
-        String comment = getParameter(request, "text-input", "");
         long timestamp = System.currentTimeMillis();
-
+        
+        // Get the input from the form.
+        String comment = getParameter(request, "text-input", ""); 
+        String numCommentsString = getParameter(request, "num-comments", "0");
+ 
         Entity commentEntity = new Entity("siteComments");
-        commentEntity.setProperty("newComment", comment);
         commentEntity.setProperty("timeStamp", timestamp);
-
+        commentEntity.setProperty("newComment", comment);
+        commentEntity.setProperty("seeNumComments", numCommentsString);
+ 
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         datastore.put(commentEntity);
-
+ 
         response.sendRedirect("/index.html");
     }
-
+ 
     /** Gets form input of specified type
     * @return request parameter or default value if parameter was not specified by client */
     private String getParameter(HttpServletRequest request, String name, String defaultValue) {
         String value = request.getParameter(name);
         if (value == null) {
+            System.out.println("Could not convert number comments wanted input to int: " + value);
```

Comment:
It looks like this debug output is now wrong?  You're not doing any conversion in this method that I can see.

Reply:
Got it! I was moving lots of debug statements around and missed this